lib/fathead/sass/parse_functions.py

```diff
@@ -0,0 +1,325 @@
+                anchor = self.parse_for_anchor(function_section[0])
+                example = self.parse_for_example(function_section[0])
+                parameter = self.parse_for_parameters(function_section[0])
+                abstract = method_signature + "<br>" + description
+                if example:
+                    description =  abstract + example
+                if parameter:
+                     abstract = "%s<br>%s"%(abstract, parameter)
+
+                url = self.create_url(anchor)
+                if function in functions:
+                    index = functions.index(function)
+                    data_elements = data[index]
+                    data_elements['abstract'] += abstract
+                else:
+                    functions.append(function)
+                    data_elements = {
+                        'function': function,
+                        'abstract': abstract,
+                        'url': url
+                    }
+                    data.append(data_elements)
+
+        self.parsed_data = data
+
+    def get_data(self):
+        """
+        Get the parsed data.
+        Returns:
+            self.parsed_data: Dict containing necessary data elements
+        """
+        return self.parsed_data
+
+
+class DataOutput(object):
+    """
+    Object responsible for outputting data into the output.txt file
+    """
+    def __init__(self, data):
+        self.data = data
+
+    def create_names_from_data(self, data_element):
+        """
+        Figure out the name of the function. Will contain the module name if one exists.
+        Args:
+            data_element: Incoming data dict
+
+        Returns:
+            Name, with whitespace stripped out
+
+        """
+        function = data_element.get('function')
+
+        dotted_name = '{}{}{}'.format(function, '.' if function  else '', function)
+        spaced_name = '{} {}'.format(function, function)
+
+        return dotted_name.strip(), spaced_name.strip()
+
+    def create_file(self):
+        """
+        Iterate through the data and create the needed output.txt file, appending to file as necessary.
+
+        """
+        with open('output.txt', 'a') as output_file:
+            for data_element in self.data:
+                if data_element.get('function'):
+                    name = data_element.get('function').encode('utf-8')
+                    abstract = data_element.get('abstract').encode('utf-8')
+                    url = data_element.get('url').encode('utf-8')
+                    
+                    list_of_data = [
+                        name,                       # unique name
+                        'A',                        # type is article
+                        name + ' function',          # redirect data
+                        '',                         # ignore
+                        'sass functions',            # no categories
```

Comment:
This should be a dynamic category based on the groupings of functions.

For example SASS has multiple `RGB` functions, including `rgb()`, `rgba()`, `blue()`, `green()`. All those related articles should have the category "sass rgb functions". An article can have multiple categories if you'd like. This helps us create groupings of Fathead articles.

That way when you search "sass rgb functions" we'll display a list of all the articles that have specified they're part of the "sass rgb functions" category.

Looking at http://sass-lang.com/documentation/Sass/Script/Functions.html#list-functions -- this is the list of `list functions`, so ideally when these articles are created they would have have the category, `sass list functions`.

Do you understand now?

If it's too difficult to determine categories we can skip this for now. It's not required, it just makes the